ERCS/erc-7811.md

```diff
@@ -0,0 +1,222 @@
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Method: `wallet_getAssets`
+
+#### Request schema
+
+```ts
+type WalletGetAssetsRequest = {
+  account: Hex;
+  requiredAssets?: Record<Hex, Hex[]>;
+};
+```
+
+`account` is a required field that indicates for which account assets are requested.
+
+`requiredAssets` is an optional field that specifies only the assets Dapp cares about on the specific chains. If it is provided, the response from the wallet SHOULD include those assets.
+
+#### Example request
+
+```json
+{
+  "account": "0x123",
+  "requiredAssets": {
+    "0x1": ["0x456", "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"],
+    "0xa": ["0x789", "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"]
+  }
+}
+```
+
+#### Response schema
+
+```ts
+type Asset = {
+  address: Hex;
+  balance: Hex;
+  type: string;
+  metadata: any;
+};
+type WalletGetAssetsResponse = Record<Hex, Asset[]>;
+```
+
+The key **SHOULD** be [EIP-155](./eip-155.md) chainId
+
+Asset fields:
+
+`address` is the address of the asset as Hex. Native assets **MUST** use `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` address specified by [ERC-7528](./eip-7528.md)
+
+`balance` is the balance of the asset as Hex
+
+**`type`:** A string indicating the type of the asset. Common asset types include but **aren’t limited to**:
+
+- **`ERC20`:** For [ERC-20](./eip-20.md) tokens
+- **`ERC721`:** For [ERC-721](./eip-721.md) tokens (NFTs)
+- **`native`:** For the chain's native currency
+
+**`metadata`:** An optional object containing additional information about the asset. The specific fields within the metadata object may vary depending on the asset type and the wallet's implementation."
+
+#### Example response
+
+```json
+{
+  "0x1": [
+    {
+      address: "0x123",
+      balance: "0xcaaea35047fe5702",
+      type: "ERC20",
+      metadata: {
+        name: "Token",
+        symbol: "TOK",
+        decimals: 18,
+      }
+    },
+    {
+      address: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
+      balance: "0xcaaea35047fe5702",
+      type: "native"
+    }
+  ],
+  "0xa": [
+    {
+      address: "0x456",
+      balance: "0xcd5595",
+      type: "ERC721",
+      metadata: {
+      //...
+      }
+    }
+  ],
+};
+```
+
+### Asset definitions
+
+This section specifies a structure for the most commonly used asset types.
+This ERC does not specify an exhaustive list of asset types. Since the type is a generic string, there could be a mismatch between the type Dapp expects and the one returned by the wallet. It’s important that no two assets share the same type. Therefore, new asset types should be specified in ERCs, either in this ERC as an amendment or in another ERC.
```

Comment:
```suggestion
This ERC does not specify an exhaustive list of asset types. Since the type is a generic string, there could be a mismatch between the type Dapp expects and the one returned by the wallet. It’s important that no two assets share the same type. Therefore, new asset types should be specified in future ERCs.
```

You cannot amend a Final ERC, so I'd recommend removing this bit.